src/mono/mono/mini/aot-compiler.c

```diff
@@ -15122,17 +15113,13 @@ aot_assembly (MonoAssembly *ass, guint32 jit_opts, MonoAotOptions *aot_options)
 		acfg->flags = (MonoAotFileFlags)(acfg->flags | MONO_AOT_FILE_FLAG_LLVM_ONLY);
 
 	acfg->assembly_name_sym = g_strdup (get_assembly_prefix (acfg->image));
-#ifdef TARGET_WASM
-	acfg->global_prefix = g_strdup_printf ("mono_aot_%s", g_strdup(mono_fixup_symbol_name (acfg->assembly_name_sym)));
-#else
```

Comment:
Reverted this changes (initially added here https://github.com/dotnet/runtime/pull/101106 ) as this is not needed for `wasm`.